sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -203,8 +203,22 @@ object GpuDeviceManager extends Logging {
       var init = RmmAllocationMode.CUDA_DEFAULT
       val features = ArrayBuffer[String]()
       if (conf.isPooledMemEnabled) {
-        init = init | RmmAllocationMode.POOL
-        features += "POOLED"
+        init = conf.rmmPool match {
+          case c if "default".equalsIgnoreCase(c) =>
+            features += "POOLED"
+            init | RmmAllocationMode.POOL
+          case c if "arena".equalsIgnoreCase(c) =>
+            features += "ARENA"
+            init | RmmAllocationMode.ARENA
+          case c if "none".equalsIgnoreCase(c) =>
+            logWarning(
+              s"RMM pooling is enabled but the pool is set to '$c', pooling is now disabled.")
+            init
+          case c =>
+            logWarning(
+              s"RMM pool set to '$c' is not supported and is being ignored.")
```

Comment:
We don't want to fail here?  The user explicitly set a config to something we don't recognize, I'm thinking this shouldn't silently progress because we know the user tried to specify something here and we're not doing it.

Reply:
Done.